### ISSUE TITLE
__adel__  destructor not called automatically

### ISSUE THREAD

**Issue by mixa2130** [mosquito/async-class#2]
Hello, hope you had a good day. I think it will be a good idea - to add self-calling class destructor

```
class Strava(AsyncClass):
    async def __ainit__(self, login, password) -> NoReturn:
        self._session = aiohttp.ClientSession()

    async def __adel__(self) -> None:
        print('hello from async del')
        # await self._session.close()

    def __del__(self):
        print('hello from del')
```

Output:
`hello from del`

As you can see - now _**` __adel__ `**_ doesn't execute - it works only through _**`close`**_ method, which is redundant for many developers

**Comment by mosquito** [comment]
that's because you redeclare __del__ method from [here](https://github.com/mosquito/async-class/blob/master/async_class.py#L138-L142)

**Comment by mixa2130** [comment]
Okay, that's an example without redeclare:

![Screenshot_20210420_155820](https://user-images.githubusercontent.com/43904727/115399971-69d9be00-a1f1-11eb-8461-90c5fc019529.png)

**Comment by tdivis** [comment]
Yes, the `__adel__` method was the reason I wanted to use this lib, it's a pity it's not called automatically :-(.

**Comment by mosquito** [comment]
@tdivis @mixa2130, this is a very experimental project, if you know what the problem is, feel free to write, I will try to investigate it, but I don't know if I can fix it in my free time or not.
